### PR TITLE
update list names

### PR DESCRIPTION
**Pull request by m-nash**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [ ] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Review by allenjzhang — APPROVED** [review]

File: samples/Azure.ResourceManager.Sample/Generated/Extensions/SubscriptionExtensions.cs [hunk 1]
```diff
@@ -1718,7 +1718,7 @@ private static UsageRestOperations GetUsageRestOperations(ClientDiagnostics clie
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <returns> A collection of resource operations that may take multiple service requests to iterate over. </returns>
         /// <exception cref="ArgumentNullException"> <paramref name="location"/> is null. </exception>
-        public static AsyncPageable<Usage> ListUsagesAsync(this SubscriptionOperations subscription, string location, CancellationToken cancellationToken = default)
+        public static AsyncPageable<Usage> GetAllUsagesAsync(this SubscriptionOperations subscription, string location, CancellationToken cancellationToken = default)
```
Comment (ShivangiReja) [inline samples/Azure.ResourceManager.Sample/Generated/Extensions/SubscriptionExtensions.cs:1721]: I think the name of this method should be `GetUsagesAsync`?

**Comment by ShivangiReja** [comment]
I think you forgot to remove `MgmtListOnly` from[ launchSettings.json](https://github.com/Azure/autorest.csharp/blob/655df2231eb0313d0e1347a3f53dfe3842f7b757/src/AutoRest.CSharp/Properties/launchSettings.json#L251-L254).

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSetContainer.cs [hunk 2]
```diff
@@ -318,59 +316,21 @@ public async virtual Task<bool> DoesExistAsync(string vmExtensionName, string ex
...
-                var response = await _restClient.ListAsync(Id.ResourceGroupName, Id.Parent.Name, Id.Name, expand, cancellationToken).ConfigureAwait(false);
-                return Response.FromValue(response.Value.Value.Select(data => new VirtualMachineExtensionVirtualMachineScaleSet(Parent, data)).ToArray() as IReadOnlyList<VirtualMachineExtensionVirtualMachineScaleSet>, response.GetRawResponse());
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
-        }
-
```
Comment (ShivangiReja) [inline samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionVirtualMachineScaleSetContainer.cs:339]: Why this method got removed from the container class and moved to the `VirtualMachineExtensionVirtualMachineScaleSetOperations.cs` class?

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionContainer.cs [hunk 3]
```diff
@@ -318,59 +316,21 @@ public async virtual Task<bool> DoesExistAsync(string vmExtensionName, string ex
...
-            {
-                var response = await _restClient.ListAsync(Id.ResourceGroupName, Id.Name, expand, cancellationToken).ConfigureAwait(false);
-                return Response.FromValue(response.Value.Value.Select(data => new VirtualMachineExtension(Parent, data)).ToArray() as IReadOnlyList<VirtualMachineExtension>, response.GetRawResponse());
-            }
-            catch (Exception e)
-            {
-                scope.Failed(e);
-                throw;
-            }
-        }
```
Comment (ShivangiReja) [inline samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionContainer.cs:338]: Why this method got removed from the container class and moved to the `VirtualMachineExtensionOperations.cs` class? This method should be generated here.

File: samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionOperations.cs [hunk 4]
```diff
@@ -374,6 +375,43 @@ public virtual Response<VirtualMachineExtension> RemoveTag(string key, Cancellat
...
+            {
+                var response = await _restClient.GetAllAsync(Id.ResourceGroupName, Id.Name, expand, cancellationToken).ConfigureAwait(false);
+                return Response.FromValue(response.Value.Value.Select(data => new VirtualMachineExtension(this, data)).ToArray() as IReadOnlyList<VirtualMachineExtension>, response.GetRawResponse());
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
```
Comment (ShivangiReja) [inline samples/Azure.ResourceManager.Sample/Generated/VirtualMachineExtensionOperations.cs:395]: This method should be generated in container class.

**Review by ShivangiReja — CHANGES_REQUESTED** [review]

File: samples/Azure.Management.Storage/Generated/Extensions/SubscriptionExtensions.cs [hunk 5]
```diff
@@ -32,19 +32,19 @@ private static StorageAccountsRestOperations GetStorageAccountsRestOperations(Cl
         /// <param name="subscription"> The <see cref="SubscriptionOperations" /> instance the method will execute against. </param>
         /// <param name="cancellationToken"> The cancellation token to use. </param>
         /// <returns> A collection of resource operations that may take multiple service requests to iterate over. </returns>
-        public static AsyncPageable<StorageAccount> ListStorageAccountsAsync(this SubscriptionOperations subscription, CancellationToken cancellationToken = default)
+        public static AsyncPageable<StorageAccount> GetStorageAccountsAsync(this SubscriptionOperations subscription, CancellationToken cancellationToken = default)
```
Comment (ArcturusZhang) [inline samples/Azure.Management.Storage/Generated/Extensions/SubscriptionExtensions.cs:35]: Should we call this `GetAllStorageAccountsAsync` to align other changes?

**Review by ArcturusZhang — COMMENTED** [review]
Could I summary this PR in the following short words?
1. Rename `List` -> `GetAll`
2. Rename `DoesExist` -> `CheckIfExist` (align with ARMCore)

File: samples/Azure.Management.Storage/Generated/Extensions/SubscriptionExtensions.cs (hunk 5 above)
Comment (m-nash) [reply samples/Azure.Management.Storage/Generated/Extensions/SubscriptionExtensions.cs:35]: this one is following
List[Noun]s -> Get[Noun]s

File: src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs [hunk 6]
```diff
@@ -109,14 +109,16 @@ public override void WriteExtension(CodeWriter writer, BuildContext<MgmtOutputLi
                         // despite that we should only have one method, but we still using an IEnumerable
                         foreach (var pagingMethod in mgmtExtensionOperation.PagingMethods)
                         {
-                            WriteExtensionPagingMethod(writer, pagingMethod.PagingResponse.ItemType, mgmtExtensionOperation.RestClient, pagingMethod, pagingMethod.Name, $"", true);
-                            WriteExtensionPagingMethod(writer, pagingMethod.PagingResponse.ItemType, mgmtExtensionOperation.RestClient, pagingMethod, pagingMethod.Name, $"", false);
+                            var methodName = /*pagingMethod.Name == "GetAll" ? "Get" :*/ pagingMethod.Name;
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs:112]: nit: can we remove these comments?

File: src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs [hunk 7]
```diff
@@ -109,14 +109,16 @@ public override void WriteExtension(CodeWriter writer, BuildContext<MgmtOutputLi
...
+                            WriteExtensionPagingMethod(writer, pagingMethod.PagingResponse.ItemType, mgmtExtensionOperation.RestClient, pagingMethod, methodName, $"", true);
+                            WriteExtensionPagingMethod(writer, pagingMethod.PagingResponse.ItemType, mgmtExtensionOperation.RestClient, pagingMethod, methodName, $"", false);
                         }
 
                         foreach (var clientMethod in mgmtExtensionOperation.ClientMethods)
                         {
-                            WriteExtensionClientMethod(writer, mgmtExtensionOperation.OperationGroup, clientMethod, clientMethod.Name, context, true, mgmtExtensionOperation.RestClient.Type.Name);
-                            WriteExtensionClientMethod(writer, mgmtExtensionOperation.OperationGroup, clientMethod, clientMethod.Name, context, false, mgmtExtensionOperation.RestClient.Type.Name);
+                            var methodName = /*clientMethod.RestClientMethod.Name == "GetAll" ? "Get" :*/ clientMethod.Name;
```
Comment (ShivangiReja) [inline src/AutoRest.CSharp/Mgmt/Generation/SubscriptionExtensionsWriter.cs:119]: nit: remove comment

**Review by ShivangiReja — APPROVED** [review]

**Pull request merged**